extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -431,6 +472,294 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
+					&& this.objectInfo.databaseScopedConfigurations[this.currentRowId].id !== 11
+					&& this.objectInfo.databaseScopedConfigurations[this.currentRowId].id !== 12
+					&& this.objectInfo.databaseScopedConfigurations[this.currentRowId].id !== 21) {
+					this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForSecondary = newValue;
+					this.dscTable.data[this.currentRowId][2] = newValue;
+				}
+				await this.updateDscTable(this.dscTable.data);
+			}
+		}, [], '', true, 150)
+		const primaryContainer = this.createLabelInputContainer(localizedConstants.ValueForPrimaryColumnHeader, this.valueForPrimaryDropdown);
+		this.dscPrimaryValueDropdownGroup = this.createGroup('', [primaryContainer], false, true);
+		await this.dscPrimaryValueDropdownGroup.updateCssStyles({ 'visibility': 'hidden' });
+
+		// Apply Primary To Secondary checkbox
+		this.setSecondaryCheckboxForDropdowns = this.createCheckbox(localizedConstants.SetSecondaryText, async (checked) => {
+			await this.dscSecondaryValueDropdownGroup.updateCssStyles({ 'visibility': checked ? 'hidden' : 'visible' });
+			this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForSecondary = checked
+				? this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForPrimary
+				: this.dscOriginalData[this.currentRowId].valueForSecondary;
+			await this.valueForSecondaryDropdown.updateProperties({ value: this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForSecondary });
+		}, true);
+		this.dscSecondaryCheckboxForDropdownGroup = this.createGroup('', [this.setSecondaryCheckboxForDropdowns], false, true);
+		await this.dscSecondaryCheckboxForDropdownGroup.updateCssStyles({ 'visibility': 'hidden' });
+
+		// Value for Secondary
+		this.valueForSecondaryDropdown = this.createDropdown(localizedConstants.ValueForSecondaryColumnHeader, async (newValue) => {
+			if (!isUndefinedOrNull(newValue) && this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForSecondary !== newValue) {
+				this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForSecondary = newValue as string;
+				this.dscTable.data[this.currentRowId][2] = newValue;
+				await this.updateDscTable(this.dscTable.data);
+			}
+		}, [], '', true, 150);
+		const secondaryContainer = this.createLabelInputContainer(localizedConstants.ValueForSecondaryColumnHeader, this.valueForSecondaryDropdown);
+		this.dscSecondaryValueDropdownGroup = this.createGroup('', [secondaryContainer], false, true);
+		await this.dscSecondaryValueDropdownGroup.updateCssStyles({ 'visibility': 'hidden' });
+
+		const valueGroup = this.createGroup('', [this.dscPrimaryValueDropdownGroup, this.dscSecondaryCheckboxForDropdownGroup, this.dscSecondaryValueDropdownGroup], true, true);
+		await valueGroup.updateCssStyles({ 'margin-left': '-10px' });
+		return valueGroup;
+	}
+
+	private async updateDscTable(data: any[][]): Promise<void> {
```

Comment:
Updating the DSC table, loads the table with new data that leads to shift the focus to first row. This causing issue when the user selects middle or down rows and make changes. Would like to hear suggestions on any alternate approaches on updating the table with live data.

Reply:
Fixed with ActiveCell property! :)